tests/dialects/test_databricks.py

```diff
@@ -32,6 +32,7 @@ def test_databricks(self):
             "CREATE TABLE foo (x INT GENERATED ALWAYS AS (YEAR(y)))",
             write={
                 "databricks": "CREATE TABLE foo (x INT GENERATED ALWAYS AS (YEAR(TO_DATE(y))))",
+                "tsql": "CREATE TABLE foo (x INTEGER AS YEAR(CAST(y AS DATE)))",
```

Comment:
Double-checking if this is correct in T-SQL..

Reply:
It looks like we need to drop `INTEGER` here, otherwise it's a syntax error, cc @dmoore247

Reply:
Just verified that the following statement is valid T-SQL.

```sql
CREATE TABLE foo (x DATETIME, y AS YEAR(CAST(x AS DATE)))
```